core/columnar/query_component.cxx

```diff
@@ -127,7 +137,23 @@ class pending_query_operation
       pending_op_ = op.value();
       return {};
     }
+    retry_timer_.cancel();
```

Comment:
An interesting situation (you can see it easily when there is a bootstrap error, but I think it applies to any situation to where creating the pending_op failed).  The situation is that http operation's callback is still "alive" at this point.  If we don't stop the timer, you see the op "timeout" later on in logs even though we never really got the query operation "off the ground".  

Also, when testing w/ the Node.js wrapper, I noticed that even if we cancel the timers, something is still holding onto the callback b/c you can see the logging continue infinitely even after the application has exited.  This infinite logging goes away with something like `invoke_callback({}, return_error);`.

Is it bad to both invoke the callback and return the same error?

Reply:
Looks good to me